### PR TITLE
ls: add support for +FORMAT in timestyle

### PR DESCRIPTION
**Pull request by dmatos2012**
Hello,
This PR addresses issue #3625, which adds support to the `+FORMAT` date in `--time-style`. Given the dynamic argument, I created a custom parsing function to accommodate for the new format. 

I have added the corresponding tests, but probably my code added might not be up to standards, as I am new at it and trying to improve, and basically doing what the borrow checker tells me every time :). I would be more than happy to make changes to improve it. 

I did notice using the +Format on the linked issue, namely `$ TIME_STYLE="+%Y-%m-%d %H:%M:%S.%N %z" ls exe -al` breaks with 
```
thread 'main' panicked at 'a Display implementation returned an error unexpectedly: Error', /rustc/fe5b13d681f25ee6474be29d748c65adcd91f69e/library/alloc/src/string.rs:2406:14
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
Basically, it fails by providing `time.format("%Y-%m-%d %H:%M:%S.%N %z" )`, but it looks more on the crate used. 

Thanks for guidance and potential feedback :)

### CODE REVIEW

**Comment by tertsdiepraam** [comment]
`%N` seems to be part that causes the panic. I'll try to make a reproducible case for the `time` crate.

**Comment by dmatos2012** [comment]
I agree on that it should not panic, however depending on the use, it would seem that the %N might not be a problem, but I am not very sure whether it would be worth to implement it. I can perhaps give it a try in a different PR, but maybe my efforts can be used on some other issues on this repo.

Btw: Pushed again to fix the `cspell` failing test.

**Comment by tertsdiepraam** [comment]
Oh yeah, `%N` is definitely something to figure out later. No need to worry about it :)
